### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by pedroque**

### CODE REVIEW

File: app/src/main/java/com/arctouch/test/ui/activity/ArcTouchActivity.kt [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            android.R.id.home -> {
+                finish()
+                true
+            }
+            else -> super.onOptionsItemSelected(item)
+        }
+    }
+
+    internal open fun setUp(savedInstanceState: Bundle?) {
+
```
Comment (arctouch-amandabrito) [inline app/src/main/java/com/arctouch/test/ui/activity/ArcTouchActivity.kt]: If you're not using these methods, they could be removed.

File: app/src/main/java/com/arctouch/test/ui/activity/MoviesActivity.kt [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+            }
+            else -> super.onOptionsItemSelected(item)
+        }
+    }
+
+    override fun setUp(savedInstanceState: Bundle?) {
+        super.setUp(savedInstanceState)
+        dataBinding = DataBindingUtil.inflate(layoutInflater, R.layout.activity_movies, null, false)
+        setContentView(dataBinding.root)
+        configViewModel = ViewModelProviders.of(this, configViewModelFactory).get(ConfigViewModel::class.java)
```
Comment (arctouch-amandabrito) [inline app/src/main/java/com/arctouch/test/ui/activity/MoviesActivity.kt]: Why use three view models instead of one?

File: app/src/main/java/com/arctouch/test/ui/activity/ArcTouchActivity.kt (hunk 1 above)
Comment (pgsroque) [reply app/src/main/java/com/arctouch/test/ui/activity/ArcTouchActivity.kt]: This is a BaseActivity, these methods are override in another activities.

File: app/src/main/java/com/arctouch/test/ui/activity/MoviesActivity.kt (hunk 2 above)
Comment (pgsroque) [reply app/src/main/java/com/arctouch/test/ui/activity/MoviesActivity.kt]: The idea is to separate each view model to make it reusable and control its own live data individually. Whenever we need config or genres they can be used in another activities just as we use in MoviesActivity and SearchMoviesActivity for example.

File: app/src/main/java/com/arctouch/test/ui/activity/ArcTouchActivity.kt (hunk 1 above)
Comment (arctouch-pietrocaselani) [reply app/src/main/java/com/arctouch/test/ui/activity/ArcTouchActivity.kt]: Ok, but why did not you use the Activity's methods `onCreate`, `onPostCreate` directly and remove those empty methods?

Comment (pgsroque) [reply app/src/main/java/com/arctouch/test/ui/activity/ArcTouchActivity.kt]: Well.. I'm used to do that just for organization purpose. As I'm used to work in a lot of projects with a lot of novice people I let theses methods organized to make other developers bind view model after set up the views. I agree that looking at it right now they can be avoided, specially the setUp method.

File: app/src/test/java/com/arctouch/test/data/repository/ConfigRepositoryImplTest.kt [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TestRule
+import org.mockito.Mock
+import org.mockito.Mockito
+import org.mockito.Mockito.verify
+import org.mockito.MockitoAnnotations
+
+class ConfigRepositoryImplTest {
```
Comment (arctouch-amandabrito) [inline app/src/test/java/com/arctouch/test/data/repository/ConfigRepositoryImplTest.kt]: Tests 👍

File: app/src/main/java/com/arctouch/test/ui/activity/MoviesActivity.kt (hunk 2 above)
Comment (pietrocaselani) [reply app/src/main/java/com/arctouch/test/ui/activity/MoviesActivity.kt]: Ok, got it, but usually the `ViewModel` reflects the state of the view. This way, it looks like your view has three states.

File: app/src/main/java/com/arctouch/test/vm/ConfigViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import com.arctouch.test.data.repository.ConfigRepository
+import com.arctouch.test.schedulers.ISchedulerProvider
+import io.reactivex.Observable
+
+class ConfigViewModel constructor(
+        private val configRepository: ConfigRepository,
+        schedulerProvider: ISchedulerProvider
+) : BaseViewModel(schedulerProvider) {
+
+    val config = MutableLiveData<Config>()
```
Comment (pietrocaselani) [inline app/src/main/java/com/arctouch/test/vm/ConfigViewModel.kt]: You were using `RxJava Observable` on all layers (Repositories, DataSources...), so why did you change to `LiveData` on the `ViewModel`?

Comment (pgsroque) [reply app/src/main/java/com/arctouch/test/vm/ConfigViewModel.kt]: To be honest I don't know. I thought it was a recommendation to use live data to communicate between view and view model and rxjava in the repository because livedata is lifecycle aware.

**Comment by pgsroque** [comment]
Thanks Amanda and Pietro for the comments!

About the crash I've tested in airplane mode and it was ok, I'll look forward to it calmly later. Thanks for reporting me.
The images are really stretched, It'd be better to be a grid to respect their resolution I guess.
